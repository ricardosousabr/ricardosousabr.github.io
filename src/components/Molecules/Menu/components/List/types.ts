export interface menuProps {
  menuItens: string[];
}
