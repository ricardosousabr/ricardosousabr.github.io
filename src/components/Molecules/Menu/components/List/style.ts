import styled from "styled-components";
import screens from "../../../../../tokens/screen";

export const BoxList = styled.ul`

  @media (min-width: ${screens.small}) {
    display: flex;
    flex-direction: row;
    justify-content: center;
  }
`
