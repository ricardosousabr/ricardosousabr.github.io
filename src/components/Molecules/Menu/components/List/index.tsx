import Button from "../../../../Atoms/Button"
import { menuProps } from "./types"
import { BoxList } from "./style"

export default function List({ menuItens }: menuProps) {

  return (
    <BoxList>
      {
        menuItens.map((itens, index) => {
          return <li key={index}><Button fontSize="16px" padding="8px 12px" background="transparent" borderRadius="8px" color="white" border="none" type="button">{itens}</Button></li>
        })
      }
    </BoxList>
  )
}
