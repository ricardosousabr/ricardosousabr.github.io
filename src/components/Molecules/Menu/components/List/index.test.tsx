import renderer from "react-test-renderer";
import List from "./index";

test("Link renders correctly", () => {
  const menuItens = ["About", "Skils", "Experience", "Education", "Contact", "Switch Theme", "Download CV"]
  const tree = renderer.create(<List menuItens={menuItens} />).toJSON();
  expect(tree).toMatchSnapshot();
});
