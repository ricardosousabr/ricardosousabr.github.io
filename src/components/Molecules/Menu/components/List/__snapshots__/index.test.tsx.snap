// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<ul
  className="sc-gtsqUy ejvOfa"
>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      About
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Skils
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Experience
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Education
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Contact
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Switch Theme
    </button>
  </li>
  <li>
    <button
      background="transparent"
      border="none"
      borderRadius="8px"
      className="sc-bdnyFh iHxJiV"
      color="white"
      padding="8px 12px"
      type="button"
    >
      Download CV
    </button>
  </li>
</ul>
`;
